decoder.go

```diff
@@ -35,49 +43,78 @@ func NewDecoder(r io.Reader) *Decoder {
 // Any error occurring mid-event is considered non-graceful and will
 // show up as some other error (most likely io.ErrUnexpectedEOF).
 func (dec *Decoder) Decode() (Event, error) {
-	// peek ahead before we start a new event so we can return EOFs
-	_, err := dec.Peek(1)
-	if err == io.ErrUnexpectedEOF {
-		err = io.EOF
-	}
-	if err != nil {
-		return nil, err
-	}
 	pub := new(publication)
 	inDecoding := false
+	var timeoutCh <-chan time.Time
+	if dec.readTimeout > 0 {
+		timeoutCh = time.After(dec.readTimeout)
```

Comment:
Why wouldn't it be GCed since there are no more references to it?

Reply:
The concern is that it is not GC'd until the timeout is expired  This is mentioned a couple of places including at  https://programming.guide/go/time-reset-wait-stop-timeout-cancel-interval.html.  I've just made a habit of always cleaning up my timers.

Reply:
Got it. I've fixed that, and also fixed a mistake in the timeout logic which made it only reset the timer when a whole event is received, rather than when any data is received (like a comment).